mlflow/models/evaluation/base.py

```diff
@@ -2050,6 +2064,18 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
             error_code=INVALID_PARAMETER_VALUE,
         )
 
+    # Monkeypatch the predict method to log traces
+    if model is not None and hasattr(model, "predict") and callable(model.predict):
+        # Store reference to original predict method to use in monkey patch
+        predict_func = model.predict
+
+        def monkey_patch_predict(x):
+            # Disable all autologging except for traces
+            with mlflow.utils.autologging_utils.restrict_langchain_autologging_to_traces_only():
+                return predict_func(x)
+
+        model.predict = monkey_patch_predict
```

Comment:
This is quite dangerous since it mutates the `model` object supplied by the user. It also doesn't support the case where an env manager is used to restore the model.